File: freeswitch/astpp-callingcards.pl
#!/usr/bin/perl -w
###########################################################################
# ASTPP - Open Source Voip Billing
# Copyright (C) 2004, Aleph Communications
#
# Contributor(s)
# "iNextrix Technologies Pvt. Ltd. - <astpp@inextrix.com>"
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details..
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>
############################################################################

use POSIX;
use Time::HiRes qw( gettimeofday tv_interval );
our $session;

use vars qw($params $ASTPP $gbl_config $gbl_astpp_db $gbl_xml_logger $gbl_xml_channels $astpp_calltype $session $sound);
                                                                             
require "/usr/local/astpp/astpp-database.pl";
require "/usr/local/astpp/astpp-logger.pl";
require "/usr/local/astpp/astpp-xml.pl";
require "/usr/local/astpp/astpp-common.pl";
require "/usr/local/astpp/astpp-callingcard-functions.pl";

#Define default variables 
$astpp_calltype = "ASTPP-CALLINGCARD";
my $sound_default_location = "/usr/local/freeswitch/sounds/en/us/callie/";


&initialize();

#Define sound files 
$sound = &define_sounds($sound_default_location);

return 1 if ( !$session->ready() );

#Welcome file playback
$session->streamFile($sound_default_location.$gbl_config->{calling_cards_welcome_file}) if($gbl_config->{calling_cards_welcome_file} ne "");

#Authenticate customer
my $cardinfo = &auth_callingcard();

#Play balance
&say_balance($cardinfo);

#Process for dialing destination number
&process_destination($cardinfo);

#IVR playback 
&playback_ivr($cardinfo);

$session->hangup();
1;

File: freeswitch/scripts/astpp-callingcards.lua
-------------------------------------------------------------------------------------
-- ASTPP - Open Source VoIP Billing Solution
--
-- Copyright (C) 2016 iNextrix Technologies Pvt. Ltd.
-- Samir Doshi <samir.doshi@inextrix.com>
-- ASTPP Version 3.0 and above
-- License https://www.gnu.org/licenses/agpl-3.0.html
--
-- This program is free software: you can redistribute it and/or modify
-- it under the terms of the GNU Affero General Public License as
-- published by the Free Software Foundation, either version 3 of the
-- License, or (at your option) any later version.
-- 
-- This program is distributed in the hope that it will be useful,
-- but WITHOUT ANY WARRANTY; without even the implied warranty of
-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-- GNU Affero General Public License for more details.
-- 
-- You should have received a copy of the GNU Affero General Public License
-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
--------------------------------------------------------------------------------------

-- Define script path 
local script_path = "/usr/share/freeswitch/scripts/astpp/";
local sound_path = "/usr/share/freeswitch/sounds/en/us/callie/"

-- Load config file
dofile("/var/lib/astpp/astpp.lua");

-- Load CONSTANT file
dofile("/usr/share/freeswitch/scripts/astpp/constant.lua");

-- Load json file to decode json string
JSON = (loadfile (script_path .."lib/JSON.lua"))();

-- Load utility file 
dofile(script_path.."lib/astpp.utility.lua");

-- Include Logger file to print messages 
dofile(script_path.."lib/astpp.logger.lua");

-- Include database connection file to connect database
dofile(script_path.."lib/astpp.db.lua");

-- Call database connection
db_connect()

-- Include common functions file 
dofile(script_path.."lib/astpp.functions.lua");
dofile(script_path.."lib/astpp.callingcard.functions.lua")
config = load_conf()

-- Include file to build xml for fs
dofile(script_path.."scripts/astpp.xml.lua");


--local test = string.gsub(debug.getinfo(1).short_src, "^(.+\\)[^\\]+$", "%1");
Logger.notice ("SECTION ")

if (not params) then
	params = {}
	function params:getHeader(name)
		self.name = name;
	end
	function params:serialize(name)
		self.name = name;
	end
end


if (config['debug']==1) then
    -- print all params 
    if (params:serialize() ~= nil) then
    	Logger.notice ("[xml_handler] Params:\n" .. params:serialize())
    end	
end

local userinfo

if (session:ready() ~= true) then
	return
end

session:setAutoHangup(false);

if(config['calling_cards_welcome_file'] ~= "") then
        session:streamFile(sound_path..config['calling_cards_welcome_file']);
else
        session:streamFile(sound_path.."astpp-welcome.wav");          
end

userinfo = auth_callingcard();

if (session:ready() ~= true) then
	return
end

Logger.notice("[Accountcode : ".. userinfo['number'] .."]" )
say_balance(userinfo)


--Process for dialing destination number
process_destination(userinfo);

--IVR playback 
playback_ivr(userinfo);

session:hangup();
